src/assignments/tech-jobs-console.rst

```diff
@@ -0,0 +1,511 @@
+   think instead about creating a nested loop to loop over each
+   ``HashMap``. If a new field is added to the job records, this approach
+   will print out the new field without any updates to ``printJobs``.
+
+Test this method before moving on to your next step:
+
+#. Save your changes.
+#. Select *Run* from the Run menu and choose to run the ``TechJobs`` class (or
+   if you have recently run it, just select the green arrow in the top right
+   corner of the screen).
+#. Select “1” to list the jobs, and then “0” to list them all.
+#. Make sure the printout matches the styling above.
+#. Test that it prints a descriptive message if no jobs are found by selecting
+   “0” to search and then “3” to search for a location. Then enter a location
+   that is not in the data (e.g. “Cancun”). Your message should be displayed.
+
+Create Method ``findByValue``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+At this stage, the application will allow users to search a *given
+column* of the data for a given String. Your next task is to enable a
+search that looks for the search term in *all* of the columns.
+
+In the ``JobData`` class, create a new (``public static``) method that
+will search for a string within each of the columns. Name it
+``findByValue``. Here are a few observations:
+
+#. The method that you write should not contain duplicate jobs. So, for
+   example, if a listing has position type “Web - Front End” and name
+   “Front end web dev” then searching for “web” should not include the
+   listing twice.
+#. As with ``printJobs``, you should write your code in a way that if a
+   new column is added to the data, your code will automatically search
+   the new column as well.
+#. You *should not* write code that calls ``findByColumnAndValue`` once
+   for each column. Rather, utilize loops and collection methods as you
+   did above.
+#. You *should*, on the other hand, read and understand
+   ``findByColumnAndValue``, since your code will look similar in some
+   ways.
+
+You’ll need to call ``findByValue`` from somewhere in ``main``. We’ll
+leave it up to you to find where. You might have noticed that when you
+try to search all columns using the app, a message is printed, so that
+is a good clue to help you find where to place this new method call.
+Once you find where to call your new method, you can *Run* the program
+again to test your code.
+
+Make Search Methods Case-Insensitive
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You’ve completed your first two tasks!
+
+Let's assume you demonstrated the updated application or the Company Team, and
+they noticed a feature that could be improved. When searching for jobs with
+the skill ``JavaScript`` some results were missing (e.g. the Watchtower
+Security job on line 31 of the CSV file). The search methods turn out to be
+case-sensitive, so they treat ``JavaScript`` and ``Javascript`` as different
+strings.
+
+The Company Team has *demanded* (ahem, *strongly requested*, they
```

Comment:
This sentence is shady.

Reply:
Added some light.  Let me know what you think.

Reply:
Awesome! Thanks!